src/views/open-api-docs-view/utils/get-request-data.ts

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+// Utils
+import {
+	getPropertyDetailPropsFromParameter,
+	getPropertyDetailPropsFromSchemaObject,
+} from './get-property-detail-props'
+// Types
+import type { PropertyDetailsProps } from '../components/property-details'
+import type { OpenAPIV3 } from 'openapi-types'
+
+/**
+ * Given OpenAPI parameter objects,
+ * Return request data formatted for display.
+ */
+export async function getRequestData(
```

Comment:
A lot of the work in `getRequestData` is being thorough in respecting OpenAPIV3 types, and collecting the data we need from various parts of the schema.